engine/launcher/src/main/scala/org/enso/launcher/upgrade/LauncherUpgrader.scala

```diff
@@ -331,7 +331,7 @@ class LauncherUpgrader(
 
       val temporaryExecutable = temporaryExecutablePath("new")
       FileSystem.copyFile(
-        extractedRoot / "bin" / OS.executableName("enso"),
+        extractedRoot / "bin" / OS.executableName(Constants.name),
```

Comment:
This change is the core of the fix - after changing the package structure, we need to use the right executable name for the copy. We forgot to update it earlier.